### ISSUE TITLE
pixi does not recognise updates to pyproject [project.scripts].

### ISSUE THREAD

**Issue by brettc** [prefix-dev/pixi#1881]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

Set up a new python project

```shell
mkdir pixi-bug
cd pixi-bug
pixi init --format pyproject
mkdir -p src/pixi_bug
touch src/pixi_bug/__init__.py
pixi install
```

Create the following file at `src/pixi_bug/hello.py`
```python
def main():
    print("Hello")

if __name__ == "__main__":
    main()
```

Test that it works (you should see "Hello" printed).
```sh
pixi run python src/pixi_bug/hello.py
```

Now try to install this as a script.
Add this section to the `pyproject.toml`
```toml
[project.scripts]
hello = "pixi_bug:hello.main"
```

Try to re-install the project and run your new script:
```sh
pixi install
pixi run hello
```
Nope, you get : `hello: command not found`

However, it works if you blow everything away and install:
```sh
rm -rf .pixi
pixi install 
pixi run hello
```

Conclusion: pixi install does not detect changes to [project.scripts] section.

### Issue description

As per title.

### Expected behavior

Changes to project.scripts should trigger a new install?

**Comment by baszalmstra** [comment]
Since we leave all of this up to `uv` is that a bug on their side @tdejager ?

**Comment by tdejager** [comment]
When trying the reproducer I think this is actually a bug on our side, because we do most of the validation logic if the correct package is installed ourself. It seems that for some reason it's not hitting validation check where it should.

**Comment by mickvangelderen** [comment]
Just ran into this. Recreating the env helped.

Aside: After browsing the [pixi.sh](pixi.sh) documentation and not finding any mention of "project.scripts", I thought maybe pixi did not support it. If this is not fixed soon, perhaps adding a few words in the documentation is worth it.

**Comment by tdejager** [comment]
Yeah, so from my testing it seems that this happens when using `run` while an install does seem to re-install the package, so I guess we need to look ad the run logic for this. 

So `project.scripts` is supported, thats just a PEP517 feature but in the case of run the package is not properly re-installed on a change even though the solve is re-triggered.

**Comment by tdejager** [comment]
Actually, after more investigation it does seem to work if we add the `--revalidate` command to run, so `pixi r --revalidate python` for example.

So to summarize:

```
pixi init --format=pyproject revalidate
cd revalidate
# Installs everything
pixi install
# This does nothing as expected
pixi install
touch pyproject.toml
# This installs (revalidate) again
pixi install

touch pyproject.toml
# Does **not** re-install (revalidate)
pixi r python
# ... while this does
pixi r --revalidate python
```

@ruben-arts  is that the behavior that we want or is this a bug?

**Comment by ruben-arts** [comment]
This has to do with the environment caching, on installation we skip the cache. So we either need to make the cache smarter or skip caching in certain scenario's. The cache currently only depends on changes in the lockfile. 

That said, we already do this when we find conda `source` packages. We could extend that with PyPI source packages too.

**Comment by baszalmstra** [comment]
I can no longer reproduce. I think this was fixed in #2886
